packages/kit/src/core/postbuild/fork.js

```diff
@@ -0,0 +1,72 @@
+import { fileURLToPath } from 'node:url';
+import child_process from 'node:child_process';
+
+/**
+ * @template T
+ * @template U
+ * @param {string} module
+ * @param {(opts: T) => U} callback
+ * @returns {(opts: T) => Promise<U>}
```

Comment:
I think this function deserves a bit of docs to understand how it works at a high level without diving into the code

```suggestion
 * Runs a task in a subprocess so any dangling stuff gets killed upon completion.
 * The subprocess needs to be the file `forked` is called in, and `forked` needs to be called eagerly at the top level.
 * @template T
 * @template U
 * @param {string} module `import.meta.url` of the file
 * @param {(opts: T) => U} callback The function that is invoked in the subprocess
 * @returns {(opts: T) => Promise<U>} A function that when called starts the subprocess
```